### PR TITLE
[EBPF] Test correct packaging of eBPF files

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

This PR is a fixed version of https://github.com/DataDog/datadog-agent/pull/25727, which had to be reverted due to some tests being broken on main. This PR changes e2e tests so that they validate that eBPF object files are correctly packaged. This means two changes:

- Add metadata to eBPF object files that helps us define the architecture they were compiled against. While eBPF assembly is architecture agnostic, offsets and structs used in them are not, which means we need some manual way to mark the architecture they were built for.
- Add code to the e2e packaging tests that ensure that system-probe is running (which means that at least one module that loaded eBPF object files was enabled) and that eBPF object files are of the correct arch.

### Motivation

Ensure that changes of the eBPF build process ([motivating PR](https://github.com/DataDog/datadog-agent/pull/25592)) have at least a basic sanity check.

### CODE REVIEW

**Comment by gjulianm** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Comment by gjulianm** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Review by vitkyrka — APPROVED** [review]

**Review by YoannGh — APPROVED** [review]

**Review by L3n41c — APPROVED** [review]
👍🏻 LGTM for files owned by @DataDog/container-integrations.

**Review by pducolin — APPROVED** [review]

**Review by spencergilbert — APPROVED** [review]
Looks fine for build and release - assuming the full pipeline's passed 👍

**Comment by gjulianm** [comment]
> Looks fine for build and release - assuming the full pipeline's passed 👍

Double checked just in case, it seems that all e2e jobs that run on main ran on this pipeline too

**Review by usamasaqib — APPROVED** [review]

File: pkg/ebpf/c/bpf_metadata.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/**
+ * This header file is included in kconfig.h and ktypes.h, which should make
```
Comment (hmahmood) [inline pkg/ebpf/c/bpf_metadata.h:2]: If this is the case, why is this being added to all the .c files in this PR? Is there a way to check if this is not included in a .c file?

Comment (gjulianm) [reply pkg/ebpf/c/bpf_metadata.h:2]: This was the comment from the previous PR, but we changed the approach and decided to include it in .c files directly. I updated the comment to reflect that. The invoke tasks check that the metadata is included after al .o files are built.

**Review by hmahmood — APPROVED** [review]

**Comment by gjulianm** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Pull request merged**